### DISCUSSION TITLE
Python程式風格規範（PEP8）

### DISCUSSION THREAD

**Discussion by chia-kai-chang** [1101-GS4719-C/Python-Community-GS4719-C#845] [Show and tell]
## 變數命名
變數名稱建議使用小寫英文字母，例如score。如果需要組合多個英文單字，可以在單字之間加上底線字元，例如：english_score。比較長的單字可以使用縮寫，例如：math_score。

如果希望變數的內容不要被修改（其他程式語言稱呼這種情況為「常數」，也就是內容不會改變的數），那麼就把這個變數名稱全部用大寫英文字母，例如MAX_NUMBER。（補充說明：Python語言並沒有常數的機制。也就是說，無法保證變數的內容不會被修改。因此就算我們把變數名稱全部用大寫英文字母，該變數的內容還是可以被修改。只不過根據Python語言的慣例，我們不應該在程式中，變更用英文大寫命名的變數內容。）

如果變數名稱包含專有名詞，例如HTTP，那就用原來的縮寫名稱，不要改成小寫，例如HTTP_port。

## Python程式的空格和空白行使用建議
在運算式中，變數、運算子和數字之間都必須用空格隔開，因為加入空格會讓運算式看起來比較清楚。

## Python程式碼的寬度
每一行程式碼的長度不要超過79個字元。如果是註解，則不要超過72個字元。如果指令超過程式碼編輯視窗的寬度，會無法完整顯示，因而造成閱讀上的麻煩。

如果遇到很長的運算式或指令，可以利用以下二種方式讓它們延續到下一行：
在圓括號、方括號和大括號中的程式碼可以隨意換行。如果是加減乘除的運算式要拆成二行，建議下一行用運算子開頭。

除了以上情況之外，必須在換行的地方加一個反斜線「\」才可以延續到下一行，而且反斜線前面要留一個空格。

## 內縮該用空格還是Tab控制字元
文字編輯程式遇到Tab控制字元就會顯示內縮的效果，可是內縮的距離和文字編輯程式的設定有關。不同的文字編輯程式遇到Tab控制字元會出現不同的內縮距離。為了讓內縮距離不要受到文字編輯程式的影響，程式碼應該使用空格做出內縮的效果，不要用Tab控制字元。

## 函式命名
函式名稱應該使用小寫英文字母，例如sum()。如果需要組合多個英文單字，可以在單字之間加上底線字元，例如：score_sum()。比較長的單字可以使用縮寫，例如：en_sum()。

如果函式名稱包含專有名詞，例如WWW，那麼就用原來的縮寫，例如connect_WWW_server()。

**Comment by williamyen0623** [comment]
變數跟函式的命名真的很重要，學習階段看不出有什麼影響，當跟別人協作的時候，沒有適當的命名是會造成很多時間的浪費的，在這個資訊爆炸的時代，時間就是產能就是金錢，這種無謂的損失應該被避免，我自己是覺得學習命名一點都不乏味，反而每次看到自己程式碼會有很療癒的感覺～
